RadixWallet/Features/ProfileBackupsFeature/RestoreProfileFromBackup/Children/RecoverWalletWithoutProfile/Coordinator/RecoverWalletWithoutProfileCoordinator.swift

```diff
@@ -131,21 +135,20 @@ public struct RecoverWalletWithoutProfileCoordinator: Sendable, FeatureReducer {
 		case let .path(.element(_, action: .importMnemonic(.delegate(delegateAction)))):
 			switch delegateAction {
 			case let .notPersisted(mnemonicWithPassphrase):
-				let mainBDFS = DeviceFactorSource.babylon(
-					mnemonicWithPassphrase: mnemonicWithPassphrase,
-					isMain: true,
-					hostInfo: .current()
-				)
-
-				let privateHD = PrivateHierarchicalDeterministicFactorSource(
-					mnemonicWithPassphrase: mnemonicWithPassphrase,
-					factorSource: mainBDFS
-				)
+				return .run { send in
+					let mainBDFS = await DeviceFactorSource.babylon(
```

Comment:
@GhenadieVP hmm it looks like Sargon does not validate / assert the length of the mnemonic for `DeviceFactorSource.babylon`? I briefly looked through the call chain and looks like no validation happens? we should probably add it, making Sargon return `Result<Self> here: https://github.com/radixdlt/sargon/blob/main/crates/sargon/src/profile/v100/factors/factor_sources/device_factor_source/device_factor_source.rs#L78-L91

Reply:
Good catch!
Did we ever have this kind of validation?